dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/listener/EnrollmentSMSListener.java

```diff
@@ -182,7 +182,7 @@ protected SMSResponse postProcess( IncomingSms sms, SMSSubmission submission )
         else
         {
             enrollment = programInstanceService.enrollTrackedEntityInstance( tei, program, enrollmentDate, incidentDate,
-                orgUnit );
+                orgUnit, enrollmentid.uid );
```

Comment:
Could we use `.getUid()` here?

Reply:
`enrollmentid` is a class called UID from the SMS compression library: https://github.com/dhis2/sms-compression/blob/master/src/main/java/org/hisp/dhis/smscompression/models/UID.java

It has a `toString()` method we can use but it can potentially return a hash of the UID if the actual UID is null.

It feels like it's probably worthwhile me adding a `getUid()` to the class.

Reply:
this is an external library (sms-compression). Another approach would be to have getUid() in the upcoming versions of sms-compression library.

Reply:
Yes let's provide public get methods for the methods we would like to callers (instead of using internal properties directly). Change the properties to be private.